src/network/networkd-dhcp4.c

```diff
@@ -949,6 +955,12 @@ static int dhcp_lease_acquired(sd_dhcp_client *client, Link *link) {
                         if (r < 0)
                                 log_link_error_errno(link, r, "Failed to set transient hostname to '%s': %m", hostname);
                 }
+
+                if (hostname && link->network->dhcp_set_static_hostname) {
+                        r = manager_set_static_hostname(link->manager, hostname);
+                        if (r < 0)
+                                log_link_error_errno(link, r, "Failed to set static hostname to '%s': %m", hostname);
+                }
```

Comment:
Is it necessary to also set the 'transient' hostname in the above? I do not think so...